gsplat/cuda/csrc/fully_fused_projection_fwd.cu

```diff
@@ -75,6 +77,9 @@ fully_fused_projection_fwd_kernel(const uint32_t C, const uint32_t N,
         quats += gid * 4;
         scales += gid * 3;
         quat_scale_to_covar_preci<T>(glm::make_vec4(quats), glm::make_vec3(scales), &covar, nullptr);
+
+        glm::mat3 rotmat = quat_to_rotmat(glm::make_vec4(quats));
+        normal = rotmat[2];
```

Comment:
Why is the normal defined like this? This is a bit weird to me. Any paper is defined in this way?

Reply:
I was loosely following the 2DGS implementation and how they read out the last column/row? on the rotation matrix.
Multiplying by the viewmatrix is not necessary because they multiply it on the python side and I felt omitting scale was natural to always read out a unit vector.

```
glm::mat3 R = quat_to_rotmat(rot);
glm::mat3 S = scale_to_mat(scale, mod);
glm::mat3 L = R * S;
normal = transformVec4x3({L[2].x, L[2].y, L[2].z}, viewmatrix);
```
https://github.com/hbb1/diff-surfel-rasterization/blob/28c928a36ea19407cd9754d068bd9a9535216979/cuda_rasterizer/forward.cu#L113

Reply:
Why the normal don't need to accumulated in rasterize_to_pixels_fwd.cu?

Reply:
They are. Normals are concatenated onto color and accumulated with them. In this PR, COLOR_DIM is 7. 3 for color, 3 for normal, and 1 for depth.
https://github.com/jefequien/gsplat-mcmc/blob/jeff/normal_consistency/gsplat/rendering.py#L320